### PR TITLE
Fix fast upsampling

### PR DESCRIPTION
**Pull request by Flix6x**
This PR fixes fast-track upsampling, which wasn't working well for data with gaps. This PR also fixes and expands the corresponding test.

### CODE REVIEW

File: timely_beliefs/beliefs/classes.py [hunk 1]
```diff
@@ -1354,13 +1368,28 @@ def resample_events(
...
-                    periods=len(df) * (self.event_resolution // event_resolution),
+                    end=df.index[-1] + self.event_resolution,
+                    closed="left",
                     freq=event_resolution,
                     name="event_start",
                 )
-                df = df.reindex(new_index).fillna(method="pad")
+                df = df.reindex(new_index).fillna(
+                    method="pad",
+                    limit=math.ceil(resample_ratio) - 1 if resample_ratio > 1 else None,
```
Comment (nhoening) [inline timely_beliefs/beliefs/classes.py:1387]: Can you comment on this?

File: timely_beliefs/tests/test_df_resampling.py [hunk 2]
```diff
@@ -241,32 +243,54 @@ def test_percentages_and_accuracy_of_probabilistic_model(df_4323: BeliefsDataFra
...
-    assert df_resampled_1.event_resolution == event_resolution_1
-    event_resolution_2 = timedelta(minutes=5)
-    df_resampled_2 = df_resampled_1.resample_events(
-        event_resolution_2, keep_only_most_recent_belief=True
+    assert df_resampled_1.event_resolution == downsampled_event_resolution
+    assert df_resampled_1["event_value"].isnull().sum() == 0
+
+    # Upsample the original frame
+    df_resampled_2 = df.resample_events(
+        upsampled_event_resolution, keep_nan_values=True
```
Comment (nhoening) [inline timely_beliefs/tests/test_df_resampling.py:266]: Is there a check if the `NaN` value is still in the result?

**Review by nhoening — APPROVED** [review]

File: timely_beliefs/tests/test_df_resampling.py (hunk 2 above)
Comment (Flix6x) [reply timely_beliefs/tests/test_df_resampling.py:266]: Actually, upon resampling there should be 2 NaN values. The relevant check is in line 273-275.

Comment (Flix6x) [reply timely_beliefs/tests/test_df_resampling.py:266]: I'll add an inline comment.

File: timely_beliefs/beliefs/classes.py (hunk 1 above)
Comment (Flix6x) [reply timely_beliefs/beliefs/classes.py:1416]: I thought it was best to explain by example. See the new inline comment.

File: timely_beliefs/tests/test_df_resampling.py (hunk 2 above)
Comment (nhoening) [reply timely_beliefs/tests/test_df_resampling.py:266]: Still confuse - Are you taking the sum of NaN values here?

**Pull request merged**